babygogo/source.py

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env python
+# Author(s): 'Tommer Wizansky' <twizansk@gmail.com>
+import abc
+import logging
+from kafka import KafkaConsumer
+
+logger = logging.getLogger(__name__)
+
+
+class Source(metaclass=abc.ABCMeta):
+    """Base class for Babygogo source classes.
+
+    The concrete source classes are responsible for retreiving incoming stream elements from
+    external systems and passing them to Stream instances for processing.
+    """
+
+    def __init__(self):
+        self._stream = None
+
+    def attach(self, stream):
+        """
+        Attach a stream to the source.  Once the source's 'start' method is called, the incoming
+        elements will be passed to the stream for processing, one at a time.
+
+        :type stream: babygogo.stream.Stream
+        """
+        self._stream = stream
+
+    @abc.abstractmethod
+    def start(self):
+        pass
+
+    def _process(self, element):
+        if self._stream is not None:
+            self._stream.process(element)
+        else:
+            logger.warn('Received element with no attached stream.  Discarding.')
+
+
+class ListSource(Source):
```

Comment:
This could be any iterable, not just a list, right? Should we call it IterableSource?